File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
from collections import Counter


def solution(want, number, discount):
    answer = 0
    dic = {v: number[i] for i, v in enumerate(want)}
    
    for i in range(len(discount) - 9):
        if dic == Counter(discount[i:i + 10]):
            answer += 1
    
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.java
import java.util.*;

class Solution {
    
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;

        Map<String, Integer> target = new HashMap<>();
        for (int i = 0; i < want.length; i++) {
            target.put(want[i], number[i]);
        }

        for (int i = 0; i <= discount.length - 10; i++) {
            Map<String, Integer> current = new HashMap<>();
            
            for (int j = i; j < i + 10; j++) {
                current.put(discount[j], current.getOrDefault(discount[j], 0) + 1);
            }

            if (current.equals(target)) {
                answer++;
            }
        }

        return answer;
    }
}